app/components/Nav/App/index.js

```diff
@@ -312,12 +313,21 @@ const App = ({ userLoggedIn }) => {
   useEffect(() => {
     if (prevNavigator.current || !navigator) return;
     const appTriggeredAuth = async () => {
-      const { PreferencesController } = Engine.context;
-      const selectedAddress = PreferencesController.state.selectedAddress;
+      const { AccountsController } = Engine.context;
+      const { selectedAccount } = AccountsController.state.internalAccounts;
+
+      const selectedInternalAccount =
+        AccountsController.state.internalAccounts.accounts[selectedAccount] ||
+        null;
+
+      const selectedAddress = selectedInternalAccount?.address;
```

Comment:
This two lines could be?
```suggestion
      const selectedAddress = AccountsController.state.internalAccounts?.accounts?.[selectedAccount]?.address;
```